src/lib.rs

```diff
@@ -553,6 +554,8 @@ pub mod tls;
 
 #[cfg(feature = "cookies")]
 mod cookies;
+mod request_ext;
```

Comment:
I wasn't sure whether to create a new file or not - I figured the request file maybe was long enough already? Would be happy to move it into that file if you want.